jaeger-core/src/main/java/com/uber/jaeger/baggage/BaggageRestrictionManager.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2017, Uber Technologies, Inc
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.uber.jaeger.baggage;
+
+public interface BaggageRestrictionManager {
+  BaggageValidity isValidBaggageKey(String key);
```

Comment:
I feel that this interface could be better by encapsulating the action to be taken when restrictions are in place. 
More concretely, I suggest replacing `isValidBaggageKey(key)` with `sanitizeBaggage(key, value)`
This abstracts an implementation detail (length of baggage) leaking out to `setBaggage`. Further, it allows for future customizability. 
This method could simply return a `Map<String,String>` and the sanitized key/value. The returned map could contain any logs that are to be added. 
On success, the map would be empty. When truncating a value, it could contain `truncated:true`, and if in the future, we decide to add other restrictions, it could easily accommodate those.